### PR TITLE
Clean code changes

### PR DESCRIPTION
**Pull request by marlop92**
1. Finished task: [BE] Improve entity model
2. Clean code changes (e.g deleting unused annotations)
3. Fix tests (they were not passing, one was only sometimes passing)
4. Change API a little bit
5. Finished task: [BE] Modify bank url approach

### CODE REVIEW

File: server/src/main/java/com/sauron/model/entities/util/TransactionDirection.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.sauron.model.entities;
```
Comment (szewa-91) [inline server/src/main/java/com/sauron/model/entities/util/TransactionDirection.java]: I don't see a point of moving it. For me, TransactionDirection is not util, it's just model.

File: server/src/main/java/com/sauron/service/CurrentBalanceServiceImpl.java [hunk 2]
```diff
@@ -32,29 +32,23 @@ public CurrentBalanceServiceImpl(UserRepository userRepository, RestTemplateBuil
...
                 .reduce(BigDecimal.ZERO, BigDecimal::add);
     }
 
-    private BigDecimal fetchCurrentBalance(Bank bankView, Long userId) {
-        return restTemplate.exchange(
-                createRequestEntity(bankView.getBalanceUrl(), userId),
-                RESPONSE_TYPE)
-                .getBody();
+    private BigDecimal fetchCurrentBalance(Bank bank, Long userId) {
+        Optional<String> apiUrl = bank.getBankApiUrl(GET_BALANCE);
```
Comment (szewa-91) [inline server/src/main/java/com/sauron/service/CurrentBalanceServiceImpl.java]: Personally, I don't like creating local Optional variables if not needed. I would inline it.

File: server/src/main/java/com/sauron/service/TransactionServiceImpl.java [hunk 3]
```diff
@@ -52,17 +52,15 @@ public TransactionServiceImpl(RestTemplateBuilder restTemplateBuilder, UserRepos
...
-                .getBody();
+    private Collection<Transaction> fetchTransactions(Bank bank, Long userId) {
+        Optional<String> apiUrl = bank.getBankApiUrl(GET_TRANSACTIONS);
+
+        Collection<Transaction> transactions = apiUrl.map(url -> restTemplate.exchange(
+                createRequestEntity(url, userId),
+                RESPONSE_TYPE).getBody())
+                .orElse(Collections.emptyList());
 
         return transactions != null ? transactions : Collections.emptyList();
```
Comment (szewa-91) [inline server/src/main/java/com/sauron/service/TransactionServiceImpl.java]: Actually, now it's never null (Optional guarantees it), so you are performing this logic twice :)

File: server/src/test/java/com/sauron/constants/BankConstants.java [hunk 4]
```diff
@@ -9,5 +16,14 @@
     public static final String MOCKED_BANK_LOGIN_URL = "http://www.mocked-bank/login";
     public static final String MOCKED_BANK_TRANSACTION_URL = "http://www.mocked-bank/transactions";
     public static final String MOCKED_BANK_BALANCE_URL = "http://www.mocked-bank/balance";
-    public static final Bank MOCKED_BANK = new Bank(MOCKED_BANK_ID, MOCKED_BANK_NAME, MOCKED_BANK_LOGIN_URL, MOCKED_BANK_TRANSACTION_URL, MOCKED_BANK_BALANCE_URL);
+
+    public static final Bank MOCKED_BANK = new Bank(MOCKED_BANK_ID, MOCKED_BANK_NAME, getMockedBankApi());
+
+    public static Map<BankApiType, String> getMockedBankApi() {
+        return Map.of(
```
Comment (szewa-91) [inline server/src/test/java/com/sauron/constants/BankConstants.java]: Cool, I didn't know this method :D

File: server/src/test/java/com/sauron/repository/BankRepositoryTestIT.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    @Test
+    public void shouldFindValidBank() {
+        Bank bank = bankRepository.getOne(1L);
+
+        then(bank).extracting(
+                Bank::getId,
+                Bank::getName
+        ).containsExactly(1L, "Full-api-bank");
+
+        then(bank.getBankApiUrl(BankApiType.DO_LOGIN)).get().isEqualTo("http://full-bank.com/login");
```
Comment (szewa-91) [inline server/src/test/java/com/sauron/repository/BankRepositoryTestIT.java]: AssertJ has assertions for Optionals, you could use them for better readibility.

File: server/src/main/java/com/sauron/model/entities/util/TransactionDirection.java (hunk 1 above)
Comment (marlop92) [reply server/src/main/java/com/sauron/model/entities/util/TransactionDirection.java]: I agree it isn't util but it isn't enity either. I moved it to just "model" package.

File: server/src/main/java/com/sauron/service/CurrentBalanceServiceImpl.java (hunk 2 above)
Comment (marlop92) [reply server/src/main/java/com/sauron/service/CurrentBalanceServiceImpl.java]: Change to both services

File: server/src/main/java/com/sauron/service/TransactionServiceImpl.java (hunk 3 above)
Comment (marlop92) [reply server/src/main/java/com/sauron/service/TransactionServiceImpl.java]: Correct. It supposed to be inside mapper function which is now fixed.

File: server/src/test/java/com/sauron/repository/BankRepositoryTestIT.java (hunk 5 above)
Comment (marlop92) [reply server/src/test/java/com/sauron/repository/BankRepositoryTestIT.java]: Fixed

File: server/src/main/java/com/sauron/model/entities/util/TransactionDirection.java (hunk 1 above)
Comment (szewa-91) [reply server/src/main/java/com/sauron/model/entities/util/TransactionDirection.java]: :+1:

**Pull request merged**